ttnn/ttnn/operations/fallback.py

```diff
@@ -0,0 +1,18 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import ttnn
+import torch
+import tt_lib
+
+THIS_MODULE = sys.modules[__name__]
+
+
+def reshape(tensor: ttnn.Tensor, shape) -> ttnn.Tensor:
+    device = tensor.device() if tensor.storage_type() == tt_lib.tensor.StorageType.DEVICE else None
```

Comment:
Calling `tensor.device()` if `storage_type()` is not `DEVICE` throws an exception.
@arakhmati we discussed this, but I am not sure if we have a decision. Why is it a good/bad idea to throw?

Reply:
It's not a good idea to throw but it's a lot of work to change this method. We should add `ttnn.get_device` that would return `Optional[ttnn.Device]`, similarly to how we added `ttnn.get_memory_config()` instead of `ttnn.memory_config()`